src/opm/parser/eclipse/EclipseState/Schedule/Well/Well.cpp

```diff
@@ -785,6 +794,21 @@ void Well::setInsertIndex(std::size_t index) {
     this->insert_index = index;
 }
 
+void Well::applyWellProdIndexScaling(const double currentEffectivePI) {
+    if (this->connections->size() == std::size_t{0})
```

Comment:
Maybe add `empty()` to `WellConnections` instead?

Reply:
Added in #2005 and used here.  I've converted this PR to draft status until #2005 is merged.